tests/stress_tests/scripts/run_memcheck.py

```diff
@@ -0,0 +1,153 @@
+            break
+
+    init_parser = argparse.ArgumentParser(add_help=False)
+    init_parser.add_argument('--timeline_report',
+                             help=f'Create timeline HTML report file name.')
+    init_parser.add_argument('--upload', action="store_true",
+                             help=f'Upload results to database.')
+    init_parser.add_argument('--gtest_parallel', help='Path to gtest-parallel to use.',
+                             default='gtest_parallel')
+    args = init_parser.parse_known_args()[0]
+
+    parser = argparse.ArgumentParser(
+        description='Run memcheck tests',
+        usage='%(prog)s [options] binary -- [additional args]',
+        parents=[init_parser])
+    parser.add_argument('binary', help='test binary to execute')
+    parser.add_argument('-d', '--output_dir',
+                        required=args.timeline_report or args.upload,
+                        help='output directory for test logs')
+    parser.add_argument('-w', '--workers', help='number of gtest-parallel workers to spawn')
+
+    parser.add_argument('--db_url',
+                        required=args.timeline_report or args.upload,
+                        help='MongoDB URL in a form "mongodb://server:port"')
+    parser.add_argument('--db_collection',
+                        required=args.timeline_report or args.upload,
+                        help=f'use collection name in {DATABASE} database',
+                        choices=COLLECTIONS)
+    parser.add_argument('--metadata',
+                        help=f'add extra runtime information, json formated')
+    parser.add_argument('--strip_log_path',
+                        metavar='REMOVE[,REPLACE]',
+                        default='',
+                        help='remove or replace parts of log path')
+
+    args = parser.parse_args()
+
+    logging.basicConfig(format="{file}: [ %(levelname)s ] %(message)s".format(
+        file=os.path.basename(__file__)), level=logging.INFO, stream=sys.stdout)
+
+    if args.output_dir:
+        if os.path.exists(args.output_dir):
+            logging.error(
+                '%s already exists. Please specify a new directory for output logs',
+                args.output_dir)
+            sys.exit(1)
+        os.makedirs(args.output_dir)
+    returncode, _ = run([args.gtest_parallel] +
+                        (['--output_dir', f'{args.output_dir}'] if args.output_dir else []) +
+                        (['--workers', f'{args.workers}'] if args.workers else []) +
+                        [args.binary] +
+                        binary_args)
+
+    if args.upload or args.timeline_report:
+        # prepare memcheck records from logs
+        metadata = json_load(args.metadata) if args.metadata else {}
+        logs = list(glob(os.path.join(args.output_dir, '**', '*.log'), recursive=True))
+        strip = args.strip_log_path.split(',') + ['']
+        records = create_memcheck_records(logs, strip[1], strip[0], append=metadata)
```

Comment:
I propose to use `get_memcheck_records(...)` from https://github.com/openvinotoolkit/openvino/pull/935/files#diff-fa316f759cd64efd53fc23e0e4943d38R64 to unify logic. After extension of this script with runs comparison, I will use this func

Reply:
this application creating records to upload from log files explicitly. Get memcheck records will involve database which is confusing. Also get_memcheck_records does not have enough arguments to create records to be uploaded. Updating it's API also unnecessary complication

Reply:
I'm working on API extension for it, but my point is in use one function with correct implementation against duplication with (maybe) different logics. We may move this implementation to some separated function like `get_memcheck_records_from_logs(...)`

Reply:
OK, maybe new function will be extra, so let's leave implementation as is